client/components/close-on-escape/test/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+import ReactDom, { unmountComponentAtNode } from 'react-dom';
+import { spy } from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import CloseOnEscape from '../';
+
+const simulateEscapeKeydown = () => (
+	document.dispatchEvent( new window.KeyboardEvent( 'keydown', { keyCode: 27 } ) )
+);
+
+describe( 'CloseOnEscape', () => {
+	describe( 'rendering', () => {
+		it( 'renders nothing', () => {
+			const wrapper = shallow( <CloseOnEscape /> );
+			expect( wrapper.type() ).to.be.a( 'null' );
+		} );
+	} );
+
+	describe( 'escape keydown event', () => {
+		let onEscapeSpy;
+		let nodes;
+
+		useFakeDom();
+
+		before( () => {
+			onEscapeSpy = spy();
+			nodes = {
+				first: document.createElement( 'div' ),
+				second: document.createElement( 'div' ),
+			};
+		} );
+
+		afterEach( () => {
+			ReactDom.unmountComponentAtNode( nodes.first );
+			ReactDom.unmountComponentAtNode( nodes.second );
+		} );
+
+		it( 'calls the `onEscape` method of stacked components in LIFO order on each escape keydown', () => {
+			ReactDom.render(
```

Comment:
I went with render() so that I could hold a reference to use for unmounting:
```js
onEscape={ function() {
  onEscapeSpy( 1 );
  unmountComponentAtNode( nodes.first ); // using the reference here
} }
```
I did this because I assumed that I either couldn't neatly pinpoint the right elements nested within a single mount( ... ) and that running 2 mounts would cause some sort of clashing (I don't really know _why_ I assumed this).

Turns out that this works a breeze:
```js
const wrapper1 = mount(
  <CloseOnEscape
    onEscape={ () => {
      onEscapeSpy( 1 );
      wrapper1.unmount();
    } }
  />
);
const wrapper1 = mount( ... );
```

It also just feels simple and easier to follow :D

Any objections to mounting 2 components? If not I'll roll with this!